### PR TITLE
CI: unpin isort 5 (#35134)

### LINKED ISSUE #35134: CI: Update to isort 5
`isort` 5.x is out with some changes to how it sorts. We'll need to update the code and unpin in our environment file.

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #35134
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ cpplint
 flake8<3.8.0
 flake8-comprehensions>=3.1.0
 flake8-rst>=0.6.0,<=0.7.0
-isort==4.3.21
+isort>=5.2.1
```
Comment (simonjayhawkins) [inline requirements-dev.txt:14]: are the changes backwards compatible or is this the minimum version we would need

Comment (VirosaLi) [reply requirements-dev.txt:14]: They fixed the conflict with MyPy check (#35134) in either version 5.2.0 or 5.2.1. So I just set it to the latest one.

**Comment by simonjayhawkins** [comment]
@TomAugspurger i'm thinking that if/when we merge this to master maybe we should backport this otherwise we could end up with some auto backports failing and needing to do them manually. wdyt?

**Comment by TomAugspurger** [comment]
Merging and backporting sounds fine.

**Comment by simonjayhawkins** [comment]
will merge this tomorrow if no objections.

**Comment by simonjayhawkins** [comment]
Thanks @VirosaLi

**Comment by simonjayhawkins** [comment]
@MeeseeksDev backport to 1.1.x

**Pull request merged**